### ISSUE TITLE
Custom dependency resolution & status of `MavenRoboSettings` 

### ISSUE THREAD

**Issue by Findns** [robolectric/robolectric#6541]
### Description

What is the current recommended way to customize dependency resolution ? More specifically, custom Maven repository [The official doc](http://robolectric.org/configuring/) mentions that the following properties can be set:
* `robolectric.dependency.repo.id` 
* `robolectric.dependency.repo.url`

The [`MavenRoboSettings`](https://github.com/robolectric/robolectric/blob/master/plugins/maven-dependency-resolver/src/main/java/org/robolectric/MavenRoboSettings.java) class specifies additional params that can be set, i.e. `username` and `password`, that are needed to access a Maven repository without anonymous access.
Currently, settings these parameters work and allow robolectric to authenticate when retrieving its dependencies.

I have a two questions:
 * Will these parameters continue to be supported? The class is marked as deprecated and they aren't mentioned in the official documentation.
 * Is there a more recommended, or more future-proof way to use an authenticated Maven repository? E.g. creating my own Test Runner or dependency resolver (and having to rely on this in all my projects) ?

Thanks in advance :bow:

**Comment by hoisie** [comment]
> Will these parameters continue to be supported? The class is marked as deprecated and they aren't mentioned in the official documentation.

Yes they will continue to be supported. However, we will need to rely on community contributions to test and fix issues, because none of the maintainers use private maven repos.

> Is there a more recommended, or more future-proof way to use an authenticated Maven repository?

I would probably suggest using a custom [PropertiesDependencyResolver](https://github.com/robolectric/robolectric/blob/master/robolectric/src/main/java/org/robolectric/internal/dependency/PropertiesDependencyResolver.java#L16). This would require a setup where Robolectric android-all jars are either embedded in or copied to the test machine. Then you have a simple list mapping SDK versions to local files. You don't have to worry about anything related to maven.

**Comment by lx0758** [comment]
```geadle
android {
    testOptions {
        // MavenRoboSettings.java
        // http://robolectric.org/configuring/#system-properties
        unitTests.all {
            systemProperty 'robolectric.dependency.repo.id', 'aliyun'
            systemProperty 'robolectric.dependency.repo.url', 'https://maven.aliyun.com/repository/public/'
            systemProperty 'robolectric.dependency.repo.username', ''
            systemProperty 'robolectric.dependency.repo.password', ''
        }
    }
}
```

<http://robolectric.org/configuring/#system-properties>
